### PR TITLE
836 handle image orientation in imagemagik

### PR DESCRIPTION
**Pull request by pixeljunkie**
| Q             | A
| ------------- | ---
| Branch        | 7.1.x
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#836
| License       | MIT

## Summary by CodeRabbit

- **New Features**
  - Implemented automatic orientation detection for image cropping.
  - Added functionality to copy image URLs to the clipboard.

- **Enhancements**
  - Improved image manipulation capabilities with new methods.

- **User Interface**
  - Updated media detail view to support new copying image URL feature.

- **Localization**
  - Added German and English translations for the new copy image URL feature.

### CODE REVIEW

**Comment by pixeljunkie** [comment]
@coderabbitai review

File: src/CoreBundle/private/library/classes/components/imageMagick/imageMagick.class.php [hunk 1]
```diff
@@ -15,52 +15,42 @@
...
-    protected $sTempDir = CMS_TMP_DIR; //'/tmp';
+    protected string $sTempDir = CMS_TMP_DIR;
 
     /**
      * quality for JPG images.
      *
-     * @var int
+     * @var int|string
      */
     protected $iJPGQuality = '90';
```
[the remark this answers is not in the document]
Comment (pixeljunkie) [reply src/CoreBundle/private/library/classes/components/imageMagick/imageMagick.class.php:36]: I know, but this is old code and i don't want to get any backwards incompatibilities by changing it. It needs deeper testing. We leave it for the moment.

File: src/CoreBundle/private/library/classes/components/imageMagick/imageMagick.class.php [hunk 2]
```diff
@@ -461,35 +432,38 @@ protected function GetNumberOfScenes()
...
         $sImParamStrip = '';
         if ($this->bStrip) {
-            $sImParamStrip = ' -strip ';
+            $sImParamStrip = ' -strip';
+        }
+        
+        $orientation = $this->getImageOrientation();
+        
+        if (null !== $orientation) {
+            $aParameter[] = $this->getRotationCommandForOrientation((string) $orientation);
```
Comment (punixde) [inline src/CoreBundle/private/library/classes/components/imageMagick/imageMagick.class.php:453]: der cast ist überflüssig, da getImageOrientation nur string oder null zurückgeben kann, und null bereits ausgeschlossen ist.

File: src/CoreBundle/private/library/classes/components/imageMagick/imageMagick.class.php [hunk 3]
```diff
@@ -519,16 +493,90 @@ public function ResizeImage($iWidth, $iHeight)
...
+    {
+        $command = null;
+        
+        switch ($orientation) {
+            case '1':
+            case 'TopLeft':
+                break;
+            case '2':
+            case 'TopRight':
+                $command = "-flop ";
```
Comment (punixde) [inline src/CoreBundle/private/library/classes/components/imageMagick/imageMagick.class.php:513]: heißt das wirklich flop und nicht auch flip? und falls ja, dann könnte man das mit dem "flip-block" zusammenführen

File: src/CoreBundle/private/library/classes/components/imageMagick/imageMagick.class.php (hunk 2 above)
Comment (pixeljunkie) [reply src/CoreBundle/private/library/classes/components/imageMagick/imageMagick.class.php:453]: You are right. Changed.

File: src/CoreBundle/private/library/classes/components/imageMagick/imageMagick.class.php (hunk 3 above)
Comment (pixeljunkie) [reply src/CoreBundle/private/library/classes/components/imageMagick/imageMagick.class.php:513]: yes. it is really flop.
"creates a horizontal mirror image by reflecting the pixels around the central y-axis"

**Pull request merged**